### DISCUSSION TITLE
ghactions usability

### DISCUSSION THREAD

**Discussion by drahnr** [mozilla/sccache#1452] [General]
#692 resulted in quite a few questions, would be great if we could move the discussion out of the issues space now that we have a discussions forum.

**Comment by aviramha** [comment]
I'm working on getting a GitHub action for that to work.
https://github.com/marketplace/actions/rust-cache-using-sccache
Currently it doesn't work because lack of binaries, I am about to fix that with providing binaries from my fork until it's officially released.

**Comment by sylvestre** [comment]
I tagged a new release but the creation of the binary is failing because of Ubuntu for now:
https://github.com/mozilla/sccache/actions/runs/3639011343/jobs/6142203376

I will retrigger it soon

**Reply by aviramha** [reply]
It still wont add the GH because the feature isn't enabled in all :) (I faced it too)
Need to edit Cargo.toml to add it to `all` features

**Reply by drahnr** [reply]
You can always install via `cargo install --features whateverextrafeatsyouneed sccache` or `cargo install --all-features sccache`

**Reply by aviramha** [reply]
> You can always install via `cargo install --features whateverextrafeatsyouneed sccache` or `cargo install --all-features sccache`

Yes but I don't want to compile on CI.

**Reply by drahnr** [reply]
I'd appreciate if you'd provide more context, I do not understand the root of your problem in regards to compiling the release tagged git referencing version of sccache with all features enabled.

**Reply by sylvestre** [reply]
@drahnr some users are doing wget of the binaries artifacts that we are generating for their CI

**Reply by drahnr** [reply]
Oh, but that's a trivial issue both to fix inside sccache and outside? Inside: https://github.com/mozilla/sccache/pull/1453

**Reply by aviramha** [reply]
Sorry, I thought it was more obvious since it's also suggested in the documentation to not build during CI:
```
If you have a Rust toolchain installed you can install sccache using cargo. Note that this will compile sccache from source which is fairly resource-intensive. For CI purposes you should use prebuilt binary packages.
```

**Reply by drahnr** [reply]
That's generally true, but you should consider building these prebuilt artifacts according to your needs and store them somewhere :) the provided ones by the sccache project is just one opinionated default set.

Regardless, @sylvestre will do another release including the GHA storage backend as part of the default set which should address the issue :)

Thank you for clarifying @aviramha !

**Reply by aviramha** [reply]
Thanks for the help and prompt responses.
sccache is next level! :)

**Reply by aviramha** [reply]
btw our new Action using our custom build is working!
Can I send a PR to suggest users to use it for GHA? (In the Readme)

**Reply by sylvestre** [reply]
Yes, please :)
I was going to ask you for this

**Reply by sylvestre** [reply]
if you could create a new doc in
https://github.com/mozilla/sccache/tree/main/docs
called "how to configure sccache with GHA storage"
it would be better

I think the readme is already too long

**Reply by aviramha** [reply]
https://github.com/mozilla/sccache/pull/1454
